src/websocket/stream.py

```diff
@@ -221,53 +197,35 @@ async def transcribe_chunk_partial(
                 result = json.dumps({"partial": text}, indent=2)
             else:
                 self.logger.error("Transcription Data is empty, no segments found")
-            await websocket.send(result)
+
+            await websocket.send_text(result)
 
             end_time = time.time()
             self.logger.debug(
                 "Partial transcription took {:.2f} s".format(end_time - start_time)
             )
+
         except Exception:
             self.logger.error(
                 "Error while transcribing audio: {}".format(traceback.format_exc())
             )
             self.close_stream = True
 
     def adjust_threshold_on_latency(self):
-        """Adjusts the partial threshold based on the latency"""
         seconds_received = len(self.overall_audio_bytes) / BYTES_PER_SECOND
         seconds_transcribed = len(self.overall_transcribed_bytes) / BYTES_PER_SECOND
         # Calculate the latency of the connection
-        seconds_difference_received_transcribed = seconds_received - seconds_transcribed
-        self.logger.info(
-            f"Latency: {seconds_difference_received_transcribed} s, Partial Threshhold {self.partial_treshold / BYTES_PER_SECOND}"
-        )
-
-        # we need to exclude the partial threshold from the calculation, because the if the threshold is reached, the latency is always bad
-        seconds_difference_received_transcribed_exclude_partial = (
-            seconds_difference_received_transcribed
-            - self.partial_treshold / BYTES_PER_SECOND
-        )
+        seconds_difference = seconds_received - seconds_transcribed
 
-        # FINAL TRANSCRIPTION TIMEOUT is out definition of bad latency
-        if (
-            seconds_difference_received_transcribed_exclude_partial
-            > FINAL_TRANSCRIPTION_TIMEOUT - 2
-        ):
-            self.logger.debug("Latency is too high, increasing partial threshold")
-            self.partial_treshold = self.partial_treshold * 1.5
-            if self.partial_treshold > self.final_treshold:
-                self.partial_treshold = self.final_treshold
-                # this is a bad case, maximum threshold
-
-        if (
-            seconds_difference_received_transcribed_exclude_partial
-            < FINAL_TRANSCRIPTION_TIMEOUT / 6
-        ):
-            self.partial_treshold = self.partial_treshold * 0.75
-            if self.partial_treshold < BYTES_PER_SECOND * PARTIAL_TRANSCRIPTION_TIMEOUT:
-                self.partial_treshold = BYTES_PER_SECOND * PARTIAL_TRANSCRIPTION_TIMEOUT
-                # Partial threshold is now equal to the default threshold
+        if seconds_difference > FINAL_TRANSCRIPTION_TIMEOUT - 2:
```

Comment:
-2 is a magic number

Reply:
![alt text](image.png)

Reply:
![image](https://github.com/user-attachments/assets/1c46ebec-d8b9-46a6-a269-865a99b0d1a4)

Unfortunately those have been magic numbers prior to our work, I do not know their exact reasoning